libvast/src/system/read_query.cpp

```diff
@@ -60,21 +60,17 @@ read_query(const invocation& inv, std::string_view file_option,
                                "unable to read from " + *fname);
       assign_query(f);
     }
-  } else if (inv.arguments.empty()) {
-    // Read query from STDIN.
-    if (::isatty(::fileno(stdout)))
-      std::cerr << "please enter a query and confirm with CTRL-D: "
-                << std::flush;
-    assign_query(std::cin);
+  } else if (inv.arguments.size() <= argument_offset) {
+    VAST_VERBOSE("not providing a query causes everything to be exported; "
+                 "please be aware that this operation may be very expensive.");
```

Comment:
I guess it doesn't hurt, but when testing the only way I was able to see this message was when I was providing a limit, ie. when I was already working around the thing it warns against :D

Reply:
You can still see it in the log file and also you can see it if you're piping the results somewhere (or writing to a file directly).